kitsune/questions/models.py

```diff
@@ -733,39 +732,15 @@ def reload_from_analytics(cls, verbose=False):
         """Update the stats from Google Analytics."""
         from kitsune.sumo import googleanalytics
 
-        today = date.today()
-        one_year_ago = today - timedelta(days=365)
-        counts = googleanalytics.pageviews_by_question(one_year_ago, today, verbose=verbose)
-        if counts:
-            # Close any existing connections because our load balancer times
-            # them out at 5 minutes and the GA calls take forever.
-            close_old_connections()
-
-            for question_id, visits in counts.items():
-                # We are trying to minimize db calls here. Let's try to update
-                # first, that will be the common case.
-                num = cls.objects.filter(question_id=question_id).update(visits=visits)
-
-                # If we were able to update, we are done.
-                if num > 0:
-                    continue
-                # If it doesn't exist yet, create it.
-                try:
-                    # For some reason unbeknowest to me,
-                    # IntegrityError doesn't get kicked up in the
-                    # QuestionVisits tests when running the tests with
-                    # a clean db. So to deal with that, we explicitly
-                    # check the db to see if the question exists.
-                    # This happens with Django 1.4.6 and South
-                    # 0.8.2. If we update either, we might want to see
-                    # if this is still a problem.
-                    Question.objects.get(pk=question_id)
-                    cls.objects.create(question_id=question_id, visits=visits)
-                except (Question.DoesNotExist, IntegrityError):
-                    # The question doesn't exist anymore, move on.
-                    continue
-        else:
-            log.warning("Google Analytics returned no interesting data," " so I kept what I had.")
+        for question_id, visits in googleanalytics.pageviews_by_question(verbose=verbose):
+            try:
+                cls.objects.update_or_create(
+                    question_id=Subquery(Question.objects.filter(id=question_id).values("id")),
+                    defaults=dict(visits=visits),
```

Comment:
Same question as above - `bulk_create` better here?

Reply:
Let me see if I can come up with a compromise.

Reply:
I'm really happy with what I came up with. See what you think. I'll test locally but with stage creds on Monday.